[camera_android] fixed (camera_android) some device crash on calling mediaRecorder.stop w…

OnClickListener2048 opened the pull request:
fixed some android devices when you calling real quick to stop camera recording, It will throw a RuntimeException instead IllegalStateException
also it fixed https://github.com/flutter/flutter/issues/163279

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

stuartmorgan commented:
Thanks for the contribution! Marking as a draft pending the completion of the rest of the checklist, as this isn't reviewable without the rest of the changes. Once it's complete, please mark as ready for review.

> * I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style), or this PR is [exempt from CHANGELOG changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog).
> * [ ]  I updated/added relevant documentation (doc comments with `///`).
> * [ ]  I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests).
> * [ ]  All existing and new tests are passing.

Also, it appears that you have commits with two different email addresses, but only signed the CLA for one of them, so you'll need to amend your commits and re-push, or sign the CLA for the other address as well.

OnClickListener2048 commented:
> Thanks for the contribution! Marking as a draft pending the completion of the rest of the checklist, as this isn't reviewable without the rest of the changes. Once it's complete, please mark as ready for review.
> 
> > * I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style), or this PR is [exempt from CHANGELOG changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog).
> > * [ ]   I updated/added relevant documentation (doc comments with `///`).
> > * [ ]   I added new tests to check the change I am making, or this PR is [test-exempt](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests).
> > * [ ]   All existing and new tests are passing.
> 
> Also, it appears that you have commits with two different email addresses, but only signed the CLA for one of them, so you'll need to amend your commits and re-push, or sign the CLA for the other address as well.

I used a diffterent email when i pushed, I'll open a new PR and close this.
